python/sdss_access/path/path.py

```diff
@@ -195,64 +246,18 @@ def extract(self, name, example):
         assert name in self.lookup_names(), '{0} must be a valid template name'.format(name)
         template = self.templates[name]
 
-        # expand the environment variable
+        # expand the environment variables
         template = os.path.expandvars(template)
 
-        # handle special functions; perform a drop in replacement
-        if re.match('%spectrodir', template):
-            template = re.sub('%spectrodir', os.environ['BOSS_SPECTRO_REDUX'], template)
-        elif re.search('%platedir', template):
-            template = re.sub('%platedir', '(.*)/{plateid:0>6}', template)
-        elif re.search('%definitiondir', template):
-            template = re.sub('%definitiondir', '{designid:0>6}', template)
-        if re.search('%plateid6', template):
-            template = re.sub('%plateid6', '{plateid:0>6}', template)
-
-        # check if template has any brackets
-        haskwargs = re.search('[{}]', template)
-        if not haskwargs:
-            return None
+        # extract keywords
+        path_dict = _extract(template, example)
+        # If we failed and we aren't strict, try again.
+        if not path_dict and not strict:
+            return _extract(os.path.basename(template), os.path.basename(example))
 
-        # escape the envvar $ and any dots
-        subtemp = template.replace('$', '\\$').replace('.', '\\.')
-        # define search pattern; replace all template keywords with regex "(.*)" group
-        research = re.sub('{(.*?)}', '(.*)', subtemp)
-        # look for matches in template and example
-        pmatch = re.search(research, template)
-        tmatch = re.search(research, example)
-
-        path_dict = {}
-        # if example match extract keys and values from the match groups
-        if tmatch:
-            values = tmatch.groups(0)
-            keys = pmatch.groups(0)
-            assert len(keys) == len(values), 'pattern and template matches must have same length'
-            parts = zip(keys, values)
-            # parse into dictionary
-            for part in parts:
-                value = part[1]
-                if re.findall('{(.*?)}', part[0]):
-                    # get the key name inside the brackets
-                    keys = re.findall('{(.*?)}', part[0])
-                    # remove the type : designation
-                    keys = [k.split(':')[0] for k in keys]
-                    # handle double bracket edge cases; remove this when better solution found
-                    if len(keys) > 1:
-                        if keys[0] == 'dr':
-                            # for {dr}{version}
```

Comment:
this chunk of code is my attempt at manually handling the `{}{}` edge cases.  It may be easier to manually catch edge cases and add each new one we discover to the test suite.  There can't be **that** many edge cases, right??

Reply:
Very few edge cases, would be my expectation also.